src/utils/starklings_directory.py

```diff
@@ -8,17 +8,13 @@
 
 
 class StarklingsDirectory:
-    def __init__(self, binary_dir_path: Path):
-        self._binary_dir_path = binary_dir_path
+    def __init__(self):
+        self._binary_dir_path = Path(__file__).parents[2].resolve()
```

Comment:
no need to check the array size here ?

Reply:
No, the __file__ will always be nested in at least 3 folders root/src/utils